analyzers/src/SonarAnalyzer.Common/Rules/Utilities/LogAnalyzerBase.cs

```diff
@@ -31,12 +31,17 @@ public abstract class LogAnalyzerBase<TSyntaxKind> : UtilityAnalyzerBase<TSyntax
         private const string Title = "Log generator";
 
         protected sealed override string FileName => "log.pb";
+        protected override bool AnalyzeGeneratedCode => true;
 
         protected LogAnalyzerBase() : base(DiagnosticId, Title) { }
 
         protected sealed override LogInfo CreateAnalysisMessage(SonarAnalysisContext context) =>
             new LogInfo { Severity = LogSeverity.Info, Text = "Roslyn version: " + typeof(SyntaxNode).Assembly.GetName().Version };
 
-        protected sealed override LogInfo CreateMessage(SyntaxTree syntaxTree, SemanticModel semanticModel) => null;
+        protected sealed override LogInfo CreateMessage(SyntaxTree syntaxTree, SemanticModel semanticModel) =>
+            Language.GeneratedCodeRecognizer.IsGenerated(syntaxTree)
```

Comment:
If the analysis time gets worse because of the fact that the IsGenerated is called now 2x much as before, we can try to use `DiagnosticAnalyzerContextHelper.IsGenerated` isntead.

Reply:
That's a good finding. We can just directly go for cached value as any other rule does. Caching should be always better than brute force